modules/web/server/src/main/scala/engage/web/server/http4s/OperationOutcome.scala

```diff
@@ -0,0 +1,30 @@
+// Copyright (c) 2016-2023 Association of Universities for Research in Astronomy, Inc. (AURA)
+// For license information see LICENSE or https://opensource.org/licenses/BSD-3-Clause
+
+package engage.web.server.http4s
+
+import cats.Eq
+import cats.syntax.eq.*
+import cats.syntax.option.*
+import io.circe.Encoder
+
+import OperationResult.*
+
+final case class OperationOutcome(
+  result: OperationResult,
+  msg:    Option[String]
+) derives Encoder.AsObject
+
+object OperationOutcome {
+
+  given Eq[OperationOutcome] = Eq.by(x => (x.result, x.msg))
```

Comment:
In case you're interested we have been using `kittens` which allows to auto derive .e.g. `Eq`